addons/pos_self_order/static/src/app/components/preset_info_popup/preset_info_popup.js

```diff
@@ -0,0 +1,114 @@
+import { Component, onWillStart, useState } from "@odoo/owl";
+import { useSelfOrder } from "@pos_self_order/app/services/self_order_service";
+import { rpc } from "@web/core/network/rpc";
+
+const { DateTime } = luxon;
+export class PresetInfoPopup extends Component {
+    static template = "pos_self_order.PresetInfoPopup";
+    static props = { callback: Function };
+
+    setup() {
+        this.selfOrder = useSelfOrder();
+        this.state = useState({
+            selectedSlot: null,
+            selectedPartnerId: null,
+            name: "",
+            phone: "",
+            street: "",
+            countryId: this.selfOrder.config.company_id.country_id.id,
+            stateId: this.selfOrder.config.company_id.country_id.state_ids[0]?.id || null,
+            city: "",
+            zip: "",
+        });
+
+        onWillStart(async () => {
+            await this.selfOrder.syncPresetSlotAvaibility(this.preset);
+        });
+    }
+
+    async setInformations() {
+        if (this.preset.needsPartner) {
+            const result = await rpc(`/pos-self-order/validate-partner`, {
+                access_token: this.selfOrder.access_token,
+                partner_id: this.state.selectedPartnerId,
+                name: this.state.name,
+                phone: this.state.phone,
+                street: this.state.street,
+                city: this.state.city,
+                country_id: this.state.countryId,
+                state_id: this.state.stateId,
+                zip: this.state.zip,
+            });
+            const partner = this.selfOrder.models.loadData(this.selfOrder.models, result);
+            localStorage.setItem("self_order_partner_id", partner["res.partner"][0].id);
```

Comment:
what's the use of saving the partner id in the localStorage?

Reply:
for the next command, it will load it and it will be available for prefil infos

Reply:
But I don't see a call to the `localStorage.getItem`. So it's not being loaded at all.